test/tools/jsontests/BlockChainTests.cpp

```diff
@@ -117,304 +120,380 @@ void doBlockchainTestNoLog(json_spirit::mValue& _v, bool _fillin)
-			continue;
+			if (_fillin)
+			{
+				o.clear(); //don't add irrelevant tests to the final file when filling
+				continue;
+			}
+			else
+			{
+				//Select test case from file by combining singleTestName and SingleTestNet
+				//singleTestName selects the fillerfile and SingleTestNet select particular test case in result test file
+				if (Options::get().singleTestName + "_" + Options::get().singleTestNet != testname)
+					continue;
+			}
 		}
 
 		BOOST_REQUIRE(o.count("genesisBlockHeader"));
 		BOOST_REQUIRE(o.count("pre"));
-		if (o.count("network"))
-			dev::test::TestBlockChain::s_sealEngineNetwork = stringToNetId(o["network"].get_str());
-
-		TestBlock genesisBlock(o["genesisBlockHeader"].get_obj(), o["pre"].get_obj());
-		if (_fillin)
-			genesisBlock.setBlockHeader(genesisBlock.blockHeader());
-
-		//TODO: genesis POW ???
-		TestBlockChain testChain(genesisBlock);
-		assert(testChain.interface().isKnown(genesisBlock.blockHeader().hash(WithSeal)));
 
 		if (_fillin)
 		{
-			o["genesisBlockHeader"] = writeBlockHeaderToJson(genesisBlock.blockHeader());
-			o["genesisRLP"] = toHex(genesisBlock.bytes(), 2, HexPrefix::Add);
-			BOOST_REQUIRE(o.count("blocks"));
+			//create a blockchain test for each network
+			for (auto& network : test::getNetworks())
+			{
+				if (!Options::get().singleTestNet.empty() && Options::get().singleTestNet != test::netIdToString(network))
+					continue;
 
-			mArray blArray;
-			size_t importBlockNumber = 0;
-			string chainname = "default";
-			string chainnetwork = "default";
-			std::map<string, ChainBranch*> chainMap = { {chainname , new ChainBranch(genesisBlock)}};
+				dev::test::TestBlockChain::s_sealEngineNetwork = network;
+				string newtestname = testname + "_" + test::netIdToString(network);
 
-			for (auto const& bl: o["blocks"].get_array())
-			{
-				mObject blObj = bl.get_obj();
-				if (blObj.count("blocknumber") > 0)
-					importBlockNumber = max((int)toInt(blObj["blocknumber"]), 1);
-				else
-					importBlockNumber++;
-
-				if (blObj.count("chainname") > 0)
-					chainname = blObj["chainname"].get_str();
-				else
-					chainname = "default";
-
-				if (blObj.count("chainnetwork") > 0)
-					chainnetwork = blObj["chainnetwork"].get_str();
-				else
-					chainnetwork = "default";
-
-				if (chainMap.count(chainname) > 0)
+				json_spirit::mObject jObj = o;
+				if (o.count("expect"))
 				{
-					if (o.count("noBlockChainHistory") == 0)
+					//prepare the corresponding expect section for the test
+					json_spirit::mArray& expects = o["expect"].get_array();
+					bool found = false;
+					for (auto& expect : expects)
 					{
-						ChainBranch::forceBlockchain(chainnetwork);
-						chainMap[chainname]->reset();
-						ChainBranch::resetBlockchain();
-						chainMap[chainname]->restoreFromHistory(importBlockNumber);
+						vector<string> netlist;
+						json_spirit::mObject& expectObj = expect.get_obj();
+						ImportTest::parseJsonStrValueIntoVector(expectObj["network"], netlist);
```

Comment:
I talked with @winsvega and the fix would complicate the code too much for this PR.